src/components/JustNotSorry.js

```diff
@@ -18,74 +19,66 @@ const WATCH_FOR_NEW_NODES = {
   childList: true,
 };
 
-class JustNotSorry extends Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      parentNode: {},
-      warnings: [],
-    };
+const JustNotSorry = ({ onEvents }) => {
+  const [state, setState] = useState({ warnings: [], parentNode: {} });
 
-    this.resetState = this.resetState.bind(this);
-    this.handleSearch = this.handleSearch.bind(this);
-    this.updateWarnings = this.updateWarnings.bind(this);
-    this.applyEventListeners = this.applyEventListeners.bind(this);
+  const resetState = () => {
+    setState({ warnings: [], parentNode: {} });
+  };
 
-    this.documentObserver = new MutationObserver(
-      forEachUniqueContentEditable(this.applyEventListeners)
-    );
-    this.documentObserver.observe(document.body, WATCH_FOR_NEW_NODES);
-  }
-
-  resetState() {
-    this.setState({ parentNode: {}, warnings: [] });
-  }
-
-  updateWarnings(email, patterns) {
+  const updateWarnings = (email, patterns) => {
     if (!email || !email.offsetParent) {
-      this.resetState();
-      return;
+      return resetState();
     }
+
     const newWarnings =
       email.childNodes.length > 0
         ? Array.from(email.childNodes)
-            .filter((node) => node.textContent !== '')
-            .flatMap((text) => findRanges(text, patterns))
+            .filter((n) => n.textContent !== '')
+            .flatMap((n) => findRanges(n, patterns))
```

Comment:
Why use `n` instead of `node` and `text`?  `n` is shorter, but I find it more difficult to understand what's being filtered and mapped.

Reply:
I originally meant to keep it as node but this was a temp rename I also forgot to revisit 😓 thanks for catching it! I also found the [createNodeIterator](https://developer.mozilla.org/en-US/docs/Web/API/Document/createNodeIterator) function so hoping this clears some of this stuff up.

Reply:
The node iterator is a nice improvement.  Thanks!